javascript/CheckForUpdates.js

```diff
@@ -0,0 +1,67 @@
+(function ($) {
+    $.entwine('ss', function ($) {
+        // p tag that holds button
+        $('#checkForUpdates').entwine({
+            // Magically set by the magic get/set{thisMemberProperty} (see poll function below)
+            PollTimeout: null,
+            onclick: function () {
+                this.setLoading();
+            },
+            onmatch: function () {
+                // Poll the current job and update the front end status
+                if (this.getButton(true).length) {
+                    this.setLoading();
+                }
+            },
+            setLoading: function () {
+                // Add warning message (set as data attribute on GridFieldRefreshButton) before
+                // first button row
+                $('.ss-gridfield-buttonrow').first().prepend(
+                    '<p class="message warning">' +
+                    this.getButton().data('message') +
+                    '</p>'
+                );
+                this.poll();
+            },
+            poll: function () {
+                let self = this;
```

Comment:
This implementation isn't using ES6 so you'll need to use `var` over `let` and `const`